### ISSUE TITLE
[Salesforce] Add the Request timeout interval for salesforce integration

### ISSUE THREAD

**Issue by milan-elastic** [elastic/integrations#7880]
# Description
In Salesforce when there is large amount of setupaudit trail events in salesforce instance then it takes more than 30 seconds to collect. 
 from Rest API. By default there is only 30 seconds timeout is configured hence the data collection is not happening and following error is getting thrown. 
Error: 
`Error while processing http request: failed to execute rf.collectResponse: failed to execute http client.Do: failed to execute http client.Do: failed to read http.response.body:`
### All changes

- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/main/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target are documented
- [x] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists, [useful guidelines](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md) to follow
- [x] At least a manual test with ES / Kibana / Agent has been performed.

**Comment by milan-elastic** [comment]
Hey @lalit-satapathy 

I am trying to finalize an approach for fixing the Timeout issue in salesforce integration. 
we can hard-code the timeout but the problem with that approach would be that if the user's data is very large then the timeout will be exceeded. 
The timeout limit for **Salesforce Rest API is 10 mins**. Please refer to [this](https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_api.htm#:~:text=The%20timeout%20limit%20for%20REST%20and%20SOAP%20API%20calls%20is%2010%20minutes).  Based on this fact, if we hardcode value of timeout to less than **10mins** then there are chances of exceeding timeout and setting the timeout to 10mins would be way too large. So we are thinking of keeping Request Timeout parameter in configuration so no user gets blocked from collecting data, We will also provide a default value to Request Timeout so that all the users do not have to deal with it unless they are running into the late response from server side.

So let me know if we should provide an request timeout option like below for the user if they still face any issue?

![image](https://github.com/elastic/integrations/assets/118723373/1c56eaf6-fbcc-4ed5-9eb9-448f2d02f9f3)

**Comment by lalit-satapathy** [comment]
>and setting the timeout to 10mins would be way too large.

- What will be the exact issue if the timeout is kept 10mins?
- Can we test with large timeout and verify if everything is working fine?

**Comment by milan-elastic** [comment]
> > and setting the timeout to 10mins would be way too large.
> 
> * What will be the exact issue if the timeout is kept 10mins?

We thought about it and the only issue would be if the timeout is 10min then if there would be any slow network issue occur then user will get notified after 10min. But if the timeout issue is lesser then we'll receive it in 5min. Comparing to the client's issue they had large amount of data but still they are able to collect in 140 seconds. So seems like 5min should be the safe timeout to consider. Also in any rare case for any of the customer if the request timeout is more then 5min then they can change the value from 5min to 10min. We can update the troubleshooting steps in readme.md for it.

> * Can we test with large timeout and verify if everything is working fine?

The issue here would be like we'll be never able to generate the data in salesforce which would take 10min timeout. So that can't be tested with the salesforce instance. But we have created the mock server using that we have tested for 10min timeout and it is working fine.
